src/SerilogWeb.Classic.WebApi/Classic/WebApi/StoreWebApInfoInHttpContextAuthenticationFilter.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+using System.Web.Http;
+using System.Web.Http.Controllers;
+using System.Web.Http.Filters;
+
+namespace SerilogWeb.Classic.WebApi
+{
+    internal class StoreWebApInfoInHttpContextAuthenticationFilter : IAuthenticationFilter
+    {
+        public bool AllowMultiple => false;
+
+        public async Task AuthenticateAsync(HttpAuthenticationContext context, CancellationToken cancellationToken)
```

Comment:
It would be better to no make the method `async` in order to avoid the async/await machinery. It's cheap but it's not free, and this will be called on every single Http request ! removing the `async` keyword and returning the `Task.FromResult(0)` should do the job.

Reply:
Oh yes, of course - thanks